cmd/nv2/gencert.go

```diff
@@ -0,0 +1,209 @@
+package main
+
+import (
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"log"
+	"math/big"
+	"net"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/urfave/cli/v2"
+)
+
+const (
+	// This needs to be configurable. Once the location of the
+	// configuration is finalized this parameter and file locations
+	// should be exposed as options on the cli with mock tests.
+	KEYS_BASE_DIR = ".notation"
+)
+
+var certificatesCommand = &cli.Command{
+	Name:  "certificates",
+	Usage: "Commands to manage certificates",
+	Subcommands: []*cli.Command{
+		generateCertCommand,
+	},
+}
+
+var generateCertCommand = &cli.Command{
+	Name:      "generate",
+	Usage:     "Generates a test crt and key file.",
+	ArgsUsage: "[host]",
+	Flags: []cli.Flag{
+		&cli.IntFlag{
+			Name:     "rsaBits",
+			Usage:    "--rsaBits 3072",
+			Required: false,
+			Value:    3072,
+		},
+		&cli.StringFlag{
+			Name:     "not-after",
+			Usage:    "--not-after 2006-01-02T15:04:05-07:00 (default is 1 year)",
+			Required: false,
+		},
+	},
+
+	Action: runGenerateCert,
+}
+
+func ensureKeysDir() (string, error) {
+
+	// Expected to ensure ~/.notation/keys
+	dirname, err := os.UserHomeDir()
```

Comment:
As discussed in #76 we should support XDG_Base_Directory.

Reply:
Notation CLI Alpha honors XDG directories.